src/linter/block_linter.go

```diff
@@ -434,6 +434,25 @@ func (b *blockLinter) checkTernary(e *ir.TernaryExpr) {
 		return // Skip `$x ?: $y` expressions
 	}
 
+	// containsParenTernary checks if a node is a parenthesized ternary operator.
+	containsParenTernary := func(n ir.Node) (containsTernary, hasBracket bool) {
+		switch n := n.(type) {
+		case *ir.ParenExpr:
+			_, containsTernary = n.Expr.(*ir.TernaryExpr)
+			return containsTernary, true
+
+		case *ir.TernaryExpr:
+			return true, false
+		}
+
+		return false, false
+	}
+
+	containsTernary, hasBracket := containsParenTernary(e.Condition)
+	if !hasBracket && containsTernary {
+		b.report(e.Condition, LevelWarning, "ternaryOrder", "in ternary operators, you must explicitly use parentheses to specify the order of operations")
+	}
```

Comment:
I may be wrong, but `containsParenTernary` could probably be removed and code may be simplified to this:
```go
_, nestedTernary := e.Condition.(*ir.TernaryExpr)
if nestedTernary {
  b.report(e.Condition, LevelWarning, "ternaryOrder", "in ternary operators, you must explicitly use parentheses to specify the order of operations")
}
```
It seems like we're only interested in nested ternary; if there is a paren expr, we'll not match it against `*ir.TernaryExpr`, so that condition seems redundant.

Reply:
You're absolutely right! Excellent simplification.